### PR TITLE
OV-7:  error handling

### PR DESCRIPTION
**Pull request by WilsonBravo**
## Test error: 

    const error: ServerValidationErrorResponse = {
        errorType: 'VALIDATION',
        message: 'Validation failed for the provided data.',
        details: [
            {
                path: ['username'],
                message: 'Username is required.'
            },
            {
                path: ['email'],
                message: 'Email must be a valid email address.'
            }
        ]
    };

## Results:
----------------------------------------------------------------
### Only message:

----------------------------------------------------------------

### CODE REVIEW

File: frontend/src/bundles/common/helpers/string-to-react-node.tsx
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/helpers/string-to-react-node.tsx:1]: helpers should not contain .tsx files. And also create a folder for file

File: frontend/src/bundles/common/middlewares/error-handling.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        if (message !== '' && !toast.isActive(toastId)) {
+            toast({
+                id: toastId,
+                title: 'An error occurred.',
+                description: stringToReactNode(message),
+                status: 'error',
+                duration: 7000,
+                isClosable: true,
+                position: 'top-right',
+            });
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/middlewares/error-handling.middleware.ts:40]: Create a separate service for Toast that we could call for multiple scenarios: error, warn, success, etc.

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/helpers/string-to-react-node.tsx
Comment (WilsonBravo) [reply frontend/src/bundles/common/helpers/string-to-react-node.tsx:1]: I deleted the helper based on the recommendations provided during the daily meeting.

File: frontend/src/bundles/common/middlewares/error-handling.middleware.ts (hunk 1 above)
Comment (WilsonBravo) [reply frontend/src/bundles/common/middlewares/error-handling.middleware.ts]: ### I implemented a toast service for success, warning, error, info, and loading messages.

File: frontend/src/bundles/common/middlewares/error-handling.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            const error = action.error as ServerValidationErrorResponse;
+            message += `${error.message}\n`;
+            if (error.details) {
+                for (const errorDetail of error.details) {
+                    message += `\t- ${errorDetail.message}\n`;
+                }
+            }
+        }
+
+        if (message !== '' && !toastService.isActive(toastId)) {
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/middlewares/error-handling.middleware.ts:27]:
```suggestion
        if (message && !toastService.isActive(toastId)) {
```

File: frontend/src/bundles/common/services/toast/toast.service.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        this.toast({
+            id: toastId,
+            title: title,
+            description: message,
+            status: 'warning',
+            duration: 7000,
+            isClosable: true,
+            position: 'top-right',
+            variant: 'solid',
+        });
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/services/toast/toast.service.ts:24]: You are using the same properties for every method except for status.

Create a separate function for calling toast and just pass status as argument

File: frontend/src/bundles/common/services/toast/toast.service.ts
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/services/toast/toast.service.ts:1]: rename it to notification.service.ts

File: frontend/src/bundles/common/services/toast/toast.service.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { type createStandaloneToast } from '@chakra-ui/react';
+
+type Constructor = {
+    toast: ReturnType<typeof createStandaloneToast>['toast'];
+};
+
+class ToastService {
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/services/toast/toast.service.ts:7]: same here

File: frontend/src/bundles/common/services/toast/toast.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import { theme } from '~/framework/theme/theme.js';
+
+import { ToastService } from './toast.service.js';
+
+const { toast } = createStandaloneToast({ theme: theme });
+
+const toastService = new ToastService({ toast });
+
+export { ToastService } from './toast.service.js';
+export { toastService };
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/services/toast/toast.ts:12]: Is it possible to combine these 2 exports?

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/services/toast/toast.ts (hunk 5 above)
Comment (WilsonBravo) [reply frontend/src/bundles/common/services/toast/toast.ts:12]: if I try to do so, I receive the next error:

Use `export…from` to re-export `NotificationService`.eslint [unicorn/prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v50.0.1/docs/rules/prefer-export-from.md)

File: frontend/src/bundles/common/services/toast/toast.service.ts (hunk 4 above)
Comment (WilsonBravo) [reply frontend/src/bundles/common/services/toast/toast.service.ts]: done

File: frontend/src/bundles/common/services/toast/toast.service.ts (hunk 3 above)
Comment (WilsonBravo) [reply frontend/src/bundles/common/services/toast/toast.service.ts]: done

File: frontend/src/bundles/common/middlewares/error-handling.middleware.ts (hunk 2 above)
Comment (WilsonBravo) [reply frontend/src/bundles/common/middlewares/error-handling.middleware.ts]: done

**Review by anton-otroshchenko — APPROVED** [review]

**Review by o-nedashkivska — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**